CineElorrietaMVC/src/main/java/Vista/PanelFacturas.java

```diff
@@ -211,38 +237,120 @@ public void actionPerformed(ActionEvent arg0) {
 				String cantidad = textCantidad.getText();
 				try {
 					producto = (String) listaProductos.getSelectedValue();
-					productoAnadir = controladorPanelFacturas.accionadoBotonAnnadirProducto(producto);
-					existeProd = true;
-				}
-				catch(Exception e) {
+					if(producto != null) {
+						existeProd = true;
+					}
+					
+				} catch (Exception e) {
 					System.out.println("No se ha seleccionado un producto");
 					lblError.setText("No se ha escogido un producto");
 				}
 				if (existeProd) {
 					try {
-						annadidos.addElement(controladorPanelFacturas.cantidadProducto(cantidad, productoAnadir));
-						textTotal.setText(controladorPanelFacturas.cantidadTotal(cantidad,  textTotal.getText(), producto));
-					}
-					catch(Exception e) {
+						if (controladorPanelFacturas.existeProducto(producto) == -1) {
+							productoAnadir = controladorPanelFacturas.accionadoBotonAnnadirProducto(producto);
```

Comment:
Aquí para añadir el producto y calcular el total podéis hacerlo solamente en una llamada. Yo le daría una vuelta a como lo estáis calculando ya que el paso de string a double con el total os está haciendo cosas raras con los decimales.

Reply:
Corregido.

Reply:
Sigue habiendo 3 llamadas al controlador, lo optimo sería en la primera de ellas que añadis el producto pasarle todo lo necesario y que te devuelvatanto el producto a añadir como la cantidad producto y cantidad total. 
Idem en las otras dos pantallas.

Reply:
Esto sigue sin resolverse, échale un ojo en las 3 pantallas.